On hover show full message

maifeeulasad opened the pull request:
<!--
What GitHub Desktop issue does this PR address? (for example, #1234)
-->

Closes #11375 

## Description
 - Added title to div, so that on hover it shows full message. Basic tooltip.

### Screenshots
![On hover show full message](https://user-images.githubusercontent.com/29339330/105168721-b5be8d80-5b44-11eb-9ee7-99d1b69e146d.gif)

<!--
If this PR touches the UI layer of the app, please include screenshots or animated gifs to show the changes.
-->



## Release notes

<!--
You can leave this blank if you're not sure.
If you don't believe this PR needs to be mentioned in the release notes, write "Notes: no-notes".
-->

Notes:

maifeeulasad committed:
#11375 - On hover show full message
 - Added title to div, so that on hover it shows full message. Basic tooltip.

sergiou87 committed:
First batch of changes to support deploying arm64 builds

sergiou87 committed:
For now only add arm64 builds for macOS to the README file

say25 commented:
I sorta think this should just wrap to multiple lines

billygriffin commented:
I'm fine with it wrapping. Would you mind updating the PR and adding another gif so we can see that before reviewing and merging?

maifeeulasad committed:
lint:fix

maifeeulasad commented:
I've used `lint:fix` this time, so all checks have been passed.

And here is another GIF showing the result : 
![https___github com_maifeeulasad_desktop_tree_enhance_clone_hover_full_message](https://user-images.githubusercontent.com/29339330/111624775-4ef0e400-8816-11eb-9b61-33bcfc437748.gif)

say25 commented:
Maybe I wasn't super clear, I think the `Receiving objects: ...`  message should be wrap to multi-lines

maifeeulasad commented:
```
  /* Limit the progress text to one line. Without
   * this the entire container will jump around in
   * y-pos due to the change in height */
   .details {
    max-width: 100%;
    @include ellipsis;
  }
```
[source](https://github.com/maifeeulasad/desktop/blob/42a5be758c2ff19ae6ea44721db3556b8c16ae7c/app/styles/ui/_cloning-repository-view.scss#L69_L71)

Should I try and change it ?

say25 commented:
Ahh that makes sense then, nevermind 😄

tidy-dev committed:
Merge branch 'development' into cherry-picking-allow-users-navigate-tabs-during-drag

sergiou87 commented:
@maifeeulasad I think it's worth changing that, or trying at least 😂 

Maybe we can move the progress bar to the center of the screen and keep it fixed. Then that `details` text goes below the progress bar, and takes as many lines as needed (up to some reasonable number of lines or height?).

Does that make sense?

tidy-dev committed:
Merge pull request #11834 from desktop/cherry-picking-allow-users-navigate-tabs-during-drag

Cherry Picking - Allow switching tab in branch menu during drag

say25 committed:
Set remote.nativeTheme.themeSource when pulling value

tidy-dev committed:
Merge pull request #11849 from desktop/cherry-picking-drag-invokes-scroll-bar

Add invoke scroll during drag event via mouse position

tidy-dev committed:
Merge pull request #11854 from desktop/bump-to-dugite-1.102.0

Bump dugite to 1.102.0

say25 committed:
Code Review Feedback, Comment Function

tidy-dev committed:
Merge pull request #11866 from desktop/cherry-picking-non-existant-file-conflict

Cherry picking non existant file conflict

sergiou87 committed:
Merge pull request #11826 from say25/themeSource-not-sticky

sergiou87 committed:
Merge pull request #11826 from say25/themeSource-not-sticky

tidy-dev committed:
Open new branch dialog during cherry-picking-flow

tidy-dev committed:
Fix merge error

Co-Authored-By: Sergio Padrino <sergio.padrino@gmail.com>

tidy-dev committed:
Release 2.7.0-beta5

Co-Authored-By: Sergio Padrino <sergio.padrino@gmail.com>

tidy-dev committed:
Typo

sergiou87 committed:
Add helper functions to expand diffs

tidy-dev committed:
Merge pull request #11870 from desktop/releases/2.7.0-beta5

tidy-dev committed:
Merge pull request #11870 from desktop/releases/2.7.0-beta5

Release 2.7.0-beta5

tidy-dev committed:
Prevent quick click bad drag state

sergiou87 committed:
Fix annoying warning

tidy-dev committed:
Add mainline 1

tidy-dev committed:
Remove dead code

tidy-dev committed:
Documenting -m 1

tidy-dev committed:
Merge pull request #11875 from desktop/cherry-picking-drag-improvements

Prevent quick click bad drag state during cherry-picking

tidy-dev committed:
Merge pull request #11875 from desktop/cherry-picking-drag-improvements

Prevent quick click bad drag state during cherry-picking

tidy-dev committed:
Update unit tests for merge commits

tidy-dev committed:
Merge pull request #11877 from desktop/cherry-picking-merge-commits-improvement

Cherry-picking use main-line 1 for merge commits

tidy-dev committed:
Merge pull request #11877 from desktop/cherry-picking-merge-commits-improvement

Cherry-picking use main-line 1 for merge commits

tidy-dev committed:
Optional on branch created callback

tidy-dev committed:
Release 2.7.0-beta6

tidy-dev committed:
Update changelog.json

Co-authored-by: Sergio Padrino <sergio.padrino@gmail.com>

tidy-dev committed:
Update changelog.json

Co-authored-by: Sergio Padrino <sergio.padrino@gmail.com>

sergiou87 committed:
Merge pull request #11872 from desktop/text-diff-expansion-helpers

Add helper functions to expand diffs

tidy-dev committed:
Merge pull request #11888 from desktop/releases/2.7.0-beta6

Releases/2.7.0 beta6

dependabot[bot] committed:
Bump y18n from 4.0.0 to 4.0.1

Bumps [y18n](https://github.com/yargs/y18n) from 4.0.0 to 4.0.1.
- [Release notes](https://github.com/yargs/y18n/releases)
- [Changelog](https://github.com/yargs/y18n/blob/master/CHANGELOG.md)
- [Commits](https://github.com/yargs/y18n/commits)

Signed-off-by: dependabot[bot] <support@github.com>

tidy-dev committed:
Release 2.7.0

tidy-dev committed:
Refactor cherry-pick retry action for checkout vs create branch

tidy-dev committed:
Rename/Refactor to reflect checkout path

tidy-dev committed:
Add path to create branch and cherry-pick

tidy-dev committed:
Make create branch return something when it succeeds

billygriffin committed:
More detail about cherry-picking

sergiou87 committed:
Merge pull request #11889 from desktop/dependabot/npm_and_yarn/y18n-4.0.1

Bump y18n from 4.0.0 to 4.0.1

sergiou87 committed:
Merge pull request #11890 from desktop/releases/2.7.0

sergiou87 committed:
Enable cherry-picking for production

sergiou87 committed:
Bump version and update changelog

sergiou87 committed:
Clean up changelog

sergiou87 committed:
Merge pull request #11895 from desktop/releases/2.7.1

sergiou87 committed:
Add diff to the state and add bottom dummy hunk

sergiou87 committed:
Create gutter markers for diff expansion

tidy-dev committed:
Refactor to a clearer - less impactful implementation

tidy-dev committed:
Tidying

sergiou87 committed:
Make expand diffs work!

sergiou87 committed:
Fix selection on expanded diffs

tidy-dev committed:
Merge branch 'development' into cherry-picking-new-branch

tidy-dev committed:
Improved new branch labeling

tidy-dev committed:
Delete branch created during cherry-picking on undo

tidy-dev committed:
Tidying

billygriffin committed:
Update roadmap

sergiou87 committed:
Don't use magic numbers!

sergiou87 committed:
Refactor hunk expansion info computation

sergiou87 committed:
Merge pull request #11899 from desktop/text-diff-bottom-dummy-hunk

sergiou87 committed:
Merge pull request #11900 from desktop/text-diff-gutter-markers

sergiou87 committed:
Merge pull request #11902 from desktop/text-diff-expansion-for-real

Diff expansion, part III: expand diffs for real now :_)

tidy-dev committed:
Merge pull request #11904 from desktop/roadmaps-gonna-roadmap

Update roadmap post-release

tidy-dev committed:
Use nullish coalescing operator

Co-authored-by: Sergio Padrino <sergio.padrino@gmail.com>

tidy-dev committed:
Checkout should by default checkout branches (not checkout during cherry-pick

tidy-dev committed:
Merge branch 'cherry-picking-new-branch' of https://github.com/desktop/desktop into cherry-picking-new-branch

sergiou87 committed:
Enable TCP trampoline for production

tidy-dev committed:
Merge pull request #11903 from desktop/cherry-picking-new-branch

Create branch during cherry-picking via the context menu

sergiou87 committed:
Merge pull request #11907 from desktop/enable-trampoline-production

Enable TCP trampoline for production

maifeeulasad commented:
I've tried line-clamping, but it didn't work for me.
My tries :
```
    display: -webkit-box;
    -webkit-box-orient: vertical;
    -webkit-line-clamp: 3;
```
And I've also tried 
```
    text-align: center;
    width: calc(100%);
    max-width: calc(100%);
```

tidy-dev committed:
Add a way to switch between English and system langauge

tidy-dev committed:
better wording

tidy-dev committed:
Make sure use can get back to English if electron is not set to English but system is

tidy-dev committed:
Merge branch 'change-spell-check-language' of https://github.com/desktop/desktop into change-spell-check-language

tidy-dev committed:
Fix cherry-pick via context menu stat tracking

tidy-dev committed:
Add tracking cherry-pick branch creation

tidy-dev committed:
Fix cherry-pick not ended on drop to current branch

tidy-dev committed:
Fix cherry-pick not ended on drop to current branch

tidy-dev committed:
Revert "Fix cherry-pick not ended on drop to current branch"

This reverts commit 9d6e36689316ce9d1a0636425753dde2ca2189e7.

tidy-dev committed:
Add some styles to make it clearer

tidy-dev committed:
Merge pull request #11914 from desktop/cherry-pick-update-stats

Cherry-pick update stats

tidy-dev committed:
Merge pull request #11916 from desktop/cherry-pick-fix-drop-on-to-current-branch-bug

Fix cherry-pick not ended on drop to current branch

tidy-dev committed:
Update hide whitespace feature flag to development

tidy-dev committed:
Merge pull request #11917 from desktop/flag-hide-white-space-for-development

Update hide whitespace feature flag to development

tidy-dev committed:
Release 2.7.2-beta1

billygriffin committed:
Update changelog.json

tidy-dev committed:
Merge pull request #11919 from desktop/releases/2.7.2-beta1

Release 2.7.2-beta1

tidy-dev committed:
Hover over pull request during drag styles

tidy-dev committed:
Track enter/leave pull request as droptarget

tidy-dev committed:
Wire up drop on pull request to dispatcher

tidy-dev committed:
Get targetBranch from PR

tidy-dev committed:
Finish cherry-pick on pr wire up

tidy-dev committed:
Fix cannot undo remote branch cherry-pick

tidy-dev committed:
Tidying

tidy-dev committed:
Tidying

tidy-dev committed:
tidying - consistent error messaging

tidy-dev committed:
Use ahead/behind to determine if branch exist on remote

say25 committed:
Split Whitespace Settings

sergiou87 committed:
Merge pull request #11906 from desktop/expanded-diff-line-selection

Diff expansion, part IV: fix selection on expanded diffs

tidy-dev committed:
Refactor to get checkout branch name on success

tidy-dev committed:
Add ending drag on pressing Escape

sergiou87 committed:
Make side-to-side diff to use text diff from state instead of props

sergiou87 committed:
Render new hunk headers in split diff

sergiou87 committed:
Add dummy hunk to expand file from the bottom in split diff

sergiou87 committed:
Expand diffs in split view!!

sergiou87 committed:
Fix selection in expanded diffs in split view

tidy-dev committed:
Refactor checkout during cherry-pick to return the checked out branch

tidy-dev committed:
Merge pull request #11933 from desktop/cherry-picking-end-drag-on-escape

Add ending drag on pressing Escape

tidy-dev committed:
Merge pull request #11912 from desktop/change-spell-check-language

Ability to change spell check language to English

tidy-dev committed:
Use gitStore.loadStatus - less expensive

tidy-dev committed:
Merge pull request #11921 from desktop/cherry-pick-fix-cannot-undo-remote-branch-cherry-pick

Fix cannot undo remote branch cherry-pick

tidy-dev committed:
Merge branch 'development' into cherry-picking-onto-prs

tidy-dev committed:
Refactor finding pull-request branch out

tidy-dev committed:
Merge pull request #11922 from desktop/cherry-picking-onto-prs

Cherry-pick on to a pull request

tidy-dev committed:
Release 2.7.2-beta2

say25 committed:
Code Clean Up

say25 committed:
enableHideWhitespaceInDiffOption in Beta

tidy-dev committed:
Update changelog.json

Co-authored-by: Billy Griffin <5091167+billygriffin@users.noreply.github.com>

tidy-dev committed:
Update changelog.json

Co-authored-by: Billy Griffin <5091167+billygriffin@users.noreply.github.com>

tidy-dev committed:
Update changelog.json

Co-authored-by: Billy Griffin <5091167+billygriffin@users.noreply.github.com>

tidy-dev committed:
Update changelog.json

Co-authored-by: Billy Griffin <5091167+billygriffin@users.noreply.github.com>

tidy-dev committed:
Update changelog.json

Co-authored-by: Billy Griffin <5091167+billygriffin@users.noreply.github.com>

billygriffin committed:
Update changelog.json

tidy-dev committed:
Merge pull request #11942 from desktop/releases/2.7.2-beta2

Release 2.7.2-beta2

tidy-dev committed:
Update changelog.json

tidy-dev committed:
Merge pull request #11944 from desktop/Remove-diff-expansion-release-note

Remove diff expansion release note

j-f1 committed:
Handle ENOTDIR when picking a clone destination

j-f1 committed:
Use a save dialog instead of an open dialog when choosing where to clone a repo

sergiou87 committed:
Merge branch 'development' into windows-arm-support

sergiou87 committed:
Merge pull request #9691 from dennisameling/windows-arm-support

sergiou87 committed:
Merge branch 'development' into fix-branch-delete-non-remote-branch-message

sergiou87 committed:
Merge pull request #11923 from desktop/fix-branch-delete-non-remote-branch-message

Delete Branch: Use ahead/behind to determine if branch exist on remote

sergiou87 committed:
Remove unused import

sergiou87 committed:
Disable updating from x64 under Rosetta to arm64

sergiou87 committed:
Rename "M1 Chip" to "Apple Silicon"

sergiou87 committed:
Fix more build issues

say25 committed:
async cleanup

say25 committed:
async clean pt2

sergiou87 committed:
Merge pull request #11926 from say25/Decouple-Whitespace-Settings

Split Whitespace Settings

tidy-dev committed:
Release 2.7.2

billygriffin committed:
Update changelog.json

billygriffin committed:
Update changelog.json

billygriffin committed:
Update changelog.json

sergiou87 committed:
Always load old and new file contents, in order to provide diff expansion

sergiou87 committed:
Fix merging adjacent hunks when expanding down

The unifiedDiffEnd value was incorrect because hunk2 in that scenario would be the adjacent hunk, and that one doesn't get its unifiedDiffEnd updated after the expansion :scream:

sergiou87 committed:
Merge pull request #11935 from desktop/hunk-expansion-type-refactor

Diff expansion, part V: refactor how we compute hunk expansion info

sergiou87 committed:
Allow expanding whole files in unified diffs

sergiou87 committed:
Allow expanding whole files in split diffs

tidy-dev committed:
Add trampoline release note

tidy-dev committed:
Merge pull request #11952 from desktop/releases/2.7.2

Release 2.7.2

sergiou87 committed:
Update app/src/ui/diff/side-by-side-diff.tsx

Co-authored-by: tidy-dev <75402236+tidy-dev@users.noreply.github.com>

sergiou87 committed:
Merge pull request #11936 from desktop/split-diff-state-refactor

Diff expansion, part VI: refactor split diff to use data from state instead of props

sergiou87 committed:
Merge pull request #11937 from desktop/split-diff-bottom-hunk

Diff expansion, part VII: add dummy hunk to allow expanding diffs from the bottom in split view

sergiou87 committed:
Merge pull request #11938 from desktop/render-split-diff-hunk-handlers

sergiou87 committed:
Merge pull request #11946 from desktop/split-diff-expand-diff

Diff expansion, part IX: expand diffs in split view 🎉

sergiou87 committed:
Merge pull request #11947 from desktop/split-diff-fix-selection

Diff expansion, part X: fix selection of expanded diffs in split view

sergiou87 committed:
Merge pull request #11954 from desktop/fix-diff-expansion-issues

sergiou87 committed:
Reset diff to restore when changing files

sergiou87 committed:
Also expand the diff when there is only one hunk at the bottom of the file

sergiou87 committed:
Merge remote-tracking branch 'origin/development' into expand-whole-files

j-f1 committed:
Fix spacing of release notes

sergiou87 committed:
Show alert when moving the app to /Applications fails

tidy-dev committed:
Merge pull request #11957 from j-f1/patch-1

Fix spacing of release notes

tidy-dev committed:
Fix snapshot and progress tracker accuracy during multiple commits

tidy-dev committed:
Snapshot should match current progress

sergiou87 committed:
Disable "Expand whole file" menu item when the file is expanded already

sergiou87 committed:
Fix issue where removed files could be expanded

sergiou87 committed:
Use `npm_config_arch` to publish the new builds

sergiou87 committed:
Use the architecture as suffix instead of path for artifacts

sergiou87 committed:
Fix upload of nuget packages

tidy-dev committed:
Parse remaining commits instead of getting range which can be erroneous

tidy-dev committed:
Tidying

sergiou87 committed:
Behind a feature flag, update x64 macOS users to the arm64 build when possible

sergiou87 committed:
Merge remote-tracking branch 'origin/development' into publish-arm64

sergiou87 committed:
Merge pull request #11683 from desktop/publish-arm64

sergiou87 committed:
Enable diff expansion for beta users

sergiou87 committed:
Merge pull request #11965 from desktop/enable-diff-expansion

Enable diff expansion for beta users

tidy-dev committed:
Rename some variables for clarity/consistency

sergiou87 committed:
Merge pull request #11955 from desktop/expand-whole-files

sergiou87 committed:
Merge pull request #11964 from desktop/cherry-picking-snapshot-bug

sergiou87 committed:
Bump version and changelog to 2.7.3-beta1

sergiou87 committed:
Fix arm64 related links from README file

sergiou87 committed:
Merge pull request #11967 from desktop/fix-readme-arm64

Fix arm64 related links from README file

sergiou87 committed:
Apply suggestions from code review

Co-authored-by: Billy Griffin <5091167+billygriffin@users.noreply.github.com>

billygriffin committed:
No working arm64 yet so removing beta link

billygriffin committed:
Merge pull request #11968 from desktop/sad-mac

Remove macOS arm64 link from README

tidy-dev committed:
Add feature flag to using save dialog

sergiou87 committed:
Remove Apple Silicon from the changelog, for now…

sergiou87 committed:
Merge pull request #11949 from j-f1/save-dialog

sergiou87 committed:
Allow Windows releases to not have a delta nuget package

sergiou87 committed:
Merge pull request #11958 from desktop/alert-error-moving-app

Show alert when moving the app to /Applications fails

sergiou87 committed:
Merge pull request #11966 from desktop/releases/2.7.3-beta1

tidy-dev commented:
@maifeeulasad I think something like this should do the trick. It gives the details all the room it would want to wrap while using height to center the message. Mind giving it a try and testing it out?
![image](https://user-images.githubusercontent.com/75402236/114168515-5596e680-98fe-11eb-86f2-5611748e8b6e.png)

sergiou87 committed:
Fix off-by-one error merging diff hunks

tidy-dev committed:
Snapshot when no sequencer files

tidy-dev committed:
Snapshot when no sequencer files

tidy-dev committed:
Refactor to use sha list arguments for cherry-pick

tidy-dev committed:
Merge pull request #11975 from desktop/fix-regression-from-snapshot-refactor

Cherry-pick Snapshot Refactor didn't account for when no sequencer files

sergiou87 committed:
Merge pull request #11972 from desktop/diff-expansion-fix-merge-hunks

Fix off-by-one error merging diff hunks

tidy-dev committed:
Remove no longer needed test

maifeeulasad commented:
@tidy-dev really appreciate your help, and I've tried it. Here is the result : 
![on overflow show full message; when tries to set line number to display, it gets overfitted](https://user-images.githubusercontent.com/29339330/114204613-338f7980-997b-11eb-99e3-174f1ae0dc5c.gif)


I surely can use **js** to calculate height of that line(as font size may vary ?) and set the height. Will that work ?

*Asking, although I haven't seen this type of code in `desktop` repository*

tidy-dev commented:
@maifeeulasad I would not be inclined to solve this with js. We prefer to manage view problems with css (if possible). It appears from your last screen capture that we are just missing the css to force a line-break on long non-breaking sections of text when needed. 

Can you add: 
overflow-wrap: break-word;

to the .details class?

![image](https://user-images.githubusercontent.com/75402236/114212622-0cf82100-9930-11eb-8a90-1156ba49217b.png)

tidy-dev committed:
Merge pull request #11976 from desktop/cherry-picking-refactor-from-range

Cherry-picking refactor to use commit list instead of range for multiple commits

sergiou87 committed:
Replace non-fatal exception with error log

sergiou87 committed:
Merge pull request #11986 from desktop/remove-trampoline-client-error

Replace non-fatal exception with error log

sergiou87 committed:
Disable diff expansion for big files

sergiou87 committed:
Merge pull request #11988 from desktop/disable-diff-expansion-big-files

Disable diff expansion for big files

sergiou87 committed:
Bump version and changelog to 2.7.3-beta2

sergiou87 committed:
(WIP) Initial work to set repository aliases

sergiou87 committed:
Many changes

sergiou87 committed:
Take alias into consideration to disambiguate repos

sergiou87 committed:
Allow removing aliases

sergiou87 committed:
Change verbiage in change alias dialog

sergiou87 committed:
Reflect alias in repo dropdown title

sergiou87 committed:
Merge pull request #11993 from desktop/releases/2.7.3-beta2

Release 2.7.3-beta2

sergiou87 committed:
Show context when creating/changing a repo alias

sergiou87 committed:
Hide repo aliases behind a feature flag

tidy-dev committed:
Show + New Branch list item during  cherry-pick drag

tidy-dev committed:
Wire it up to the create branch dialog

tidy-dev committed:
Update windows ghost text

tidy-dev committed:
Update new branch margin

sergiou87 committed:
Merge pull request #12000 from desktop/repository-alias

sergiou87 committed:
Allow users to skip moving the app to /Applications

sergiou87 committed:
Cleanup strings

tidy-dev committed:
Merge pull request #12001 from desktop/create_branch_during_drag_n_drop_cherry_pick

Cherry-pick: create branch during drag n drop

sergiou87 committed:
Merge pull request #12006 from desktop/move-to-applications-checkbox

maifeeulasad commented:
@tidy-dev this doesn't solve the line count problem. 
And I'm quite sure, *I need* to use `js` to do so. Maybe there is some way around with `css`, but I'm not that good in `css`.

youholemy commented:
Preschool

youholemy commented:
Dims

sergiou87 commented:
> this doesn't solve the line count problem

@maifeeulasad I'm sorry, what's that problem exactly? It seems to me that @tidy-dev's last suggestion should solve the problem of long paths. Isn't that right? 🤔

dependabot[bot] committed:
Bump ssri from 6.0.1 to 6.0.2

Bumps [ssri](https://github.com/npm/ssri) from 6.0.1 to 6.0.2.
- [Release notes](https://github.com/npm/ssri/releases)
- [Changelog](https://github.com/npm/ssri/blob/v6.0.2/CHANGELOG.md)
- [Commits](https://github.com/npm/ssri/compare/v6.0.1...v6.0.2)

Signed-off-by: dependabot[bot] <support@github.com>

sergiou87 committed:
Merge pull request #12024 from desktop/dependabot/npm_and_yarn/ssri-6.0.2

Bump ssri from 6.0.1 to 6.0.2

sergiou87 committed:
Use italics for aliases in the repository list

sergiou87 committed:
Add the repo alias to its tooltip

sergiou87 committed:
Merge pull request #12027 from desktop/repo-alias-styling

Use italics for aliases in the repository list

sergiou87 committed:
Remove `enableCreateForkFlow`

sergiou87 committed:
Remove `enableAutomaticGitProxyConfiguration`

sergiou87 committed:
Merge pull request #12028 from desktop/repo-alias-tooltip

Add the repo alias to its tooltip

sergiou87 committed:
Remove `enableCreateGitHubIssueFromMenu`

sergiou87 committed:
Remove `enableUpdateRemoteUrl`

sergiou87 committed:
Remove `enableForkyCreateBranchUI`

sergiou87 committed:
Remove `enableGitTagsDisplay`

sergiou87 committed:
Remove `enableGitTagsCreation`

sergiou87 committed:
Remove `enableForkSettings`

sergiou87 committed:
Remove `enableDiscardLines`

sergiou87 committed:
Remove `enableSideBySideDiffs`

sergiou87 committed:
Remove `enableDefaultBranchSetting`

sergiou87 committed:
Remove `enableCherryPicking`

maifeeulasad commented:
> (up to some reasonable number of lines or height?).

as mentioned https://github.com/desktop/desktop/pull/11424#issuecomment-804208358, or should I just implement it, from @tidy-dev 's solution

tidy-dev commented:
@maifeeulasad If the concern is allowing it to go all the way down to the bottom of the screen, we can force some margin. This doesn't limit to a specific line count, but I think the goal should just be to allow it to wrap as many times as it needs without it starting to look weird by going out the bottom of the screen.. and potentially pushing the text up. 

For that... 
```
.details {
    max-width: 100%;
    height: 30vh;
    margin-bottom: 20vh;
    overflow-wrap: anywhere;
    overflow: hidden;
    text-overflow: ellipsis;
}
```

Below I use zoomed in as much as allowed and shorted the screen as much as allowed while provide an extreme amount of text. This still looks fine to me :) 
![image](https://user-images.githubusercontent.com/75402236/115411651-a68dc100-a1c1-11eb-9e68-b437001afcc2.png)


Here is what that looks back on 100% zoom and 1100px height.
![image](https://user-images.githubusercontent.com/75402236/115411776-c6bd8000-a1c1-11eb-9267-60c4aa2ccc04.png)

The one pitfall here is that the text-overflow:ellipsis doesn't apply most of the time in this scenario... But, my assumption is that most use cases will have 2 and at most 3 lines here and will never hit the max-height here.

sergiou87 committed:
Merge pull request #12029 from desktop/feature-flag-cleanup

sergiou87 committed:
Bump version and changelog to 2.7.3-beta3

sergiou87 committed:
Apply suggestions from code review

Co-authored-by: Billy Griffin <5091167+billygriffin@users.noreply.github.com>

maifeeulasad committed:
gave it some fixed position(no more jumping around) and height, solution by @tidy-dev

maifeeulasad committed:
gave it some fixed position(no more jumping around) and height, solution by @tidy-dev

memodrnn commented on app/styles/ui/_cloning-repository-view.scss:
```diff
@@ -66,12 +66,13 @@
     overflow: hidden;
```
…

maifeeulasad replied:
hello @memodrnn , i couldn't get you. a little explanation would be helpful

sergiou87 replied:
Don't worry about that comment @maifeeulasad, it was just a spammy message 😓

vidalvenegas2 commented on app/src/ui/cloning-repository.tsx:
```diff
@@ -27,7 +27,9 @@ export class CloningRepositoryView extends React.Component<
         </div>
         <progress value={progressValue} />
-        <div className="details">{this.props.progress.description}</div>
+        <div title={this.props.progress.description} className="details">
```
```suggestion
        <div title={this.props.progress.description} className="details">
```

sergiou87 committed:
Merge pull request #12031 from desktop/releases/2.7.3-beta3

tidy-dev commented:
@maifeeulasad Thank you for all the time you have put into this! 🙇  Do you think you could also update the branch to the latest that is in development? After that we will can look at getting this merged. 🙂

maifeeulasad committed:
Merge branch 'enhance/clone_hover_full_message' of https://github.com/maifeeulasad/desktop into enhance/clone_hover_full_message

maifeeulasad commented:
@tidy-dev updated branch from upstream/development.

tidy-dev commented:
@maifeeulasad Hmm.. seems something went a little wrong with the merge -> It is showing 291 file changes. Think you can try creating a new branch off latest development and open a new pr with just your changes. :)

tidy-dev commented:
Closing this in favor of #12035
